internal/clients/composer/client.go

```diff
@@ -8,25 +8,22 @@ import (
 	"fmt"
 	"io"
 	"net/http"
-	"net/url"
 	"os"
 	"path/filepath"
 	"strings"
-	"sync"
 
 	"github.com/google/uuid"
-	"github.com/sirupsen/logrus"
+	"github.com/osbuild/image-builder/internal/oauth2"
 )
 
 type ComposerClient struct {
 	composerURL string
 
 	tokenURL     string
 	offlineToken string
-	accessToken  string
 	clientId     string
 	clientSecret string
-	tokenMu      sync.RWMutex
+	Tokener      oauth2.Tokener
 
 	client *http.Client
 }
```

Comment:
Client no longer needs anything else than:

* a URL
* tokener
* http.Client

As tokener will hold the rest (client id/secret, token URL, offline/access token).